compiler/src/middle_end/optimize_dead_assignments.re

```diff
@@ -1,16 +1,29 @@
 open Anftree;
 open Grain_typed;
 
-let used_symbols = ref(Ident.empty: Ident.tbl(bool));
+let used_symbols = ref(Ident.empty: Ident.tbl(unit));
 
-let mark_used = id => used_symbols := Ident.add(id, true, used_symbols^);
+let mark_used = id => used_symbols := Ident.add(id, (), used_symbols^);
 
 let get_comp_purity = c =>
   Option.value(~default=false) @@ Analyze_purity.comp_expression_purity(c);
 
+let safe_to_remove_import = i =>
+  switch (i.imp_desc) {
+  | GrainValue(_) => true
+  | _ => false
+  };
+
 let can_remove = (ident, value) =>
-  try(!Ident.find_same(ident, used_symbols^)) {
-  | Not_found => get_comp_purity(value)
+  switch (Ident.find_same_opt(ident, used_symbols^)) {
+  | Some () => false
```

Comment:
I'd honestly prefer if this tbl didn't use unit so the Reason code didn't look weird. It could even just be `0` or something

Reply:
Unit feels the most semantic since we're using the hash table here more as a hash "set", but yeah it totally makes the code look a little off 😛 

WDYT about just doing `Some(_)` here?

Reply:
I like that better.